code/game/gamemodes/dynamic/dynamic_unfavorable_situation.dm

```diff
@@ -8,12 +8,9 @@
 	INVOKE_ASYNC(src, PROC_REF(_unfavorable_situation))
 
 /datum/game_mode/dynamic/proc/_unfavorable_situation()
-	var/static/list/unfavorable_random_events = list(
-		/datum/round_event_control/immovable_rod,
-		/datum/round_event_control/meteor_wave,
-		/datum/round_event_control/portal_storm_syndicate,
-	)
-
+	var/static/list/unfavorable_random_events = list()
+	if (!length(unfavorable_random_events))
+		unfavorable_random_events = generate_unfavourable_events()
```

Comment:
```suggestion
	var/static/list/unfavorable_random_events = list()
	if (!length(unfavorable_random_events))
		unfavorable_random_events = generate_unfavourable_events()
```

Why even have this check if you clear out the list before you check length?

Reply:
It doesn't empty the list because it's static, once it has been declared it will just keep its value between proc calls.
`generate_unfavourable_events` will only be run the first time this proc is called.

Reply:
okay if you know what you're doing it's fine then i was just confused